docs/dev/tombstone.md

```diff
@@ -0,0 +1,572 @@
+
+### Partition Tombstone
+
+```CQL
+DELETE FROM ks.tbl WHERE pk = 'partition tombstone';
+
+SELECT * from MUTATION_FRAGMENTS(ks.tbl) WHERE pk = 'partition tombstone';
+
+ pk                  | mutation_source | partition_region | ck1 | ck2 | position_weight | metadata                                                                            | mutation_fragment_kind | value
+---------------------+-----------------+------------------+-----+-----+-----------------+-------------------------------------------------------------------------------------+------------------------+-------
+ partition tombstone |      memtable:0 |                0 |     |     |                 | {"tombstone":{"timestamp":1743054972857790,"deletion_time":"2025-03-27 05:56:12z"}} |        partition start |  null
+ partition tombstone |      memtable:0 |                3 |     |     |                 |                                                                                null |          partition end |  null
+
+(2 rows)
+```
+
+### Range Tombstone
+
+Delete a range:
+```CQL
+DELETE FROM ks.tbl WHERE pk = 'range tombstone 1' AND ck1 = 0 AND ck2 > 100 AND ck2 < 200;
+
+SELECT * FROM MUTATION_FRAGMENTS(ks.tbl) WHERE pk = 'range tombstone 1';
+
+ pk                | mutation_source | partition_region | ck1 | ck2 | position_weight | metadata                                                                            | mutation_fragment_kind | value
+-------------------+-----------------+------------------+-----+-----+-----------------+-------------------------------------------------------------------------------------+------------------------+-------
+ range tombstone 1 |      memtable:0 |                0 |     |     |                 |                                                                    {"tombstone":{}} |        partition start |  null
+ range tombstone 1 |      memtable:0 |                2 |   0 | 100 |               1 | {"tombstone":{"timestamp":1743055013006807,"deletion_time":"2025-03-27 05:56:53z"}} | range tombstone change |  null
+ range tombstone 1 |      memtable:0 |                2 |   0 | 200 |              -1 |                                                                    {"tombstone":{}} | range tombstone change |  null
+ range tombstone 1 |      memtable:0 |                3 |     |     |                 |                                                                                null |          partition end |  null
+
+(4 rows)
+```
+
+Internally, range tombstones are represented by so-called "range tombstone change" objects.
+There are two such objects: one at the start of the range tombstone (the lower bound key) and one at the end of the range tombstone (the higher bound key).
+The range tombstone end object resets the tombstone to the null-tombstone (no deletion).
```

Comment:
There are also range tombstones with only one bound.

Reply:
Only if two range tombstones overlap. I will add an overlap case too.

Reply:
Done in v2.